### ISSUE TITLE
Unhandled WebSocketConnectionClosedException

### ISSUE THREAD

**Issue by h4nsu** [polkascan/py-substrate-interface#383]
Occasionally this line:
https://github.com/polkascan/py-substrate-interface/blob/84485c1f7774016ed2f93380b3a6ec041c28b2c1/substrateinterface/base.py#L329 
blows up with `WebSocketConnectionClosedException`. Would it be ok to wrap it with an auto-reconnect try-catch just like here:
https://github.com/polkascan/py-substrate-interface/blob/84485c1f7774016ed2f93380b3a6ec041c28b2c1/substrateinterface/base.py#L271-L282

**Comment by arjanz** [comment]
Yes handling this in the application is the preferred way. There is an 'auto_reconnect' feature built in (`auto_reconnect=True` during init), which will work as well, but will hide side-effects like subscriptions being reset from the application.

**Comment by h4nsu** [comment]
Sorry, what I wrote was confusing. Here's what I meant:

This:
https://github.com/polkascan/py-substrate-interface/blob/84485c1f7774016ed2f93380b3a6ec041c28b2c1/substrateinterface/base.py#L272
is wrapped with a try-block and handles broken sockets transparently when `auto_reconnect=True`.

Shouldn't the same practice be applied here:
https://github.com/polkascan/py-substrate-interface/blob/84485c1f7774016ed2f93380b3a6ec041c28b2c1/substrateinterface/base.py#L329
so that I don't need to reconnect manually in my application and I can simply rely on `auto_reconnect` doing what it promises? ;)
